### DISCUSSION TITLE
Comments in wildcard files

### DISCUSSION THREAD

**Discussion by masslevel** [adieyal/sd-dynamic-prompts#224] [Ideas]
Is there any functionality that would allow comments in wildcard files that would be excluded from populating Dynamic Prompt constructs?

I'm looking into creating more complex prompt builder constructs and setups. A file header and / or comments in a wildcard file would make it easier to keep track of changes and updates. 

I'm not sure if more standardized or unique comment formats would make more sense should any kind of syntax functionality be added in the future. Although multi-replacing comment blocks isn't a big problem should the comment formatting need to be adjusted in the future.

### Common comment formatting
```
// comment
# comment
/* comment */
```

### Multiline
```
// multi-line
// comment

# multi-line
# comment

/*
    multi-line
    comment
*/
```

I'm not sure how complicated it would be to implement this. Also if inline comments would be possible. I guess the whitespace character before the comment would need to be considered.

### Inline comments
```
spring // blooming
summer // sunny
fall // windy
winter // cold
```

Wasn't there a feature a couple of days ago that allowed comments in the prompt field?

Thank you!

**Comment by akx** [comment]
Sure! Empty lines and lines starting with the octothorpe (`#`) are already ignored in wildcard files.

https://github.com/adieyal/dynamicprompts/blob/add30353808ca239d58c831b4b921481caedab76/src/dynamicprompts/wildcardfile.py#L23
https://github.com/adieyal/dynamicprompts/blob/add30353808ca239d58c831b4b921481caedab76/src/dynamicprompts/utils.py#L8-L9

**Reply by masslevel** [reply]
Ah - sorry, I've missed that. Thank you! I was only looking for "comments" in README.md and the UI settings.

**Comment by Woisek** [comment]
Talking about wildcards: Is the "combination term" {word1|word2|...} not respected in a wildcard?
I'm currently developing a prompt and put some combinations into wildcard files, but when it gets selected, it stays at {...|...}.
Anything I missed here?

Edit:
An example:
RAW photo of (a girl:1.2) inside a {clean|gritty} kitchen, full body shot, platinum blonde fringe cut, (high detailed face:1.3), (high detailed skin:1.2), detailed skin, skin imperfections, olive skin, normal clothes, Kodak Funsaver disposable camera, sharp focus, film grain, Fujifilm XT3, dynamic lighting, shadows, (photo taken at dawn, side lit:1.2)

The {clean|gritty} was not resolved in the prompt.
